### ISSUE TITLE
 [a11y][Mobile] Alert dialog should announce "Alert"

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#64926]
b/166127321

According to the GM specification, on Android, When the alert dialog appears, screen readers should announce the alert with below a11y label: **alert** {title} {dialog content}.

Currently, the AlertDialog doesn't announce "alert" by default. Create a simple alert dialog via:

```dart
Widget _buildAlertDialog() {
    return AlertDialog(
      title: Semantics(
          header: true,
          child: Text(
            'Dialog Title',
          )),
      content: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
              'This is a sample dialog to demonstrate the AlertDialog behavior. '),
        ],
      ),
      actions: <Widget>[
        FlatButton(
          onPressed: _onPressed,
          child: const Text('DISAGREE'),
        ),
        FlatButton(
          onPressed: _onPressed,
          child: const Text('AGREE'),
        ),
      ],
      scrollable: true,
    );
  }
```

**Comment by TahaTesser** [comment]
<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: _buildAlertDialog(),
      ),
    );
  }

  Widget _buildAlertDialog() {
    return AlertDialog(
      title: Semantics(
          header: true,
          child: Text(
            'Dialog Title',
          )),
      content: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
              'This is a sample dialog to demonstrate the AlertDialog behavior. '),
        ],
      ),
      actions: <Widget>[
        FlatButton(
          onPressed: () {},
          child: const Text('DISAGREE'),
        ),
        FlatButton(
          onPressed: () {},
          child: const Text('AGREE'),
        ),
      ],
      scrollable: true,
    );
  }
}


```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.139, on Mac OS X 10.15.6 19G2021,
    locale en-GB)
    • Flutter version 1.22.0-2.0.pre.139 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision c0ea00ed3f (7 hours ago), 2020-08-30 22:20:16 -0700
    • Engine revision bada9fc5f3
    • Dart version 2.10.0 (build 2.10.0-76.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (5 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios
      • iOS 13.6.1
    • iPhone 11 (mobile)     • A63D6F51-59F3-4A75-961F-3769709F3A92 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64
      • Mac OS X 10.15.6 19G2021
    • Web Server (web)       • web-server                           •
      web-javascript • Flutter Tools
    • Chrome (web)           • chrome                               •
      web-javascript • Google Chrome 84.0.4147.135

• No issues found!
```

</details>

**Comment by HansMuller** [comment]
CC @chunhtai

**Comment by chunhtai** [comment]
I am not aware of a way to make talkback to treat the pop up as a native alert dialog, I have posted a question in our internal Q&A

question id 4730682291650560

Worst case we can manually add an "Alert" string when we send TYPE_WINDOW_STATE_CHANGED event to android, but that is last resort because we will then have to worry about localization.

**Comment by Hixie** [comment]
Looks like we have "alertDialogLabel" which is already localized.

**Comment by chunhtai** [comment]
We can reused the label which is good.

However, I figured we don't have the ability to add namesRoute that will only be announce once when route changes

For example if we have following node
```dart
Semantics(
   label: 'Alert',
   namesRoute: true,
   child: Text('dialog title')
)
```
When the route changes, it will announce `Alert, dialog title`. However, if you moving the focus to some where and move it back to this node it will still announce `Alert, dialog title`.

The native behavior is that it will announce `Alert, dialog title` when route change, but only announce `dialog title` when the focus moved around and moved back. I think we will need a new API in semantics to add a pre fix route name that will only be announced when the route change and ignored when a node is focused.

**Comment by chunhtai** [comment]
PR reverted due to post submit test failure
